velox/functions/prestosql/Repeat.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+// See documentation at https://prestodb.io/docs/current/functions/array.html
+class RepeatFunction : public exec::VectorFunction {
+ public:
+  static constexpr int32_t kMaxResultEntries = 10000;
+
+  bool isDefaultNullBehavior() const override {
+    // For repeat(null, count), it should output an array with
+    // repeating null for "count" times.
+    return false;
+  }
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    const auto numRows = rows.size();
+    auto pool = context.pool();
+    auto count = 0;
+    DecodedVector* countDecoded = nullptr;
+
+    if (args[1]->isConstantEncoding()) {
+      if (args[1]->as<ConstantVector<int32_t>>()->isNullAt(0)) {
+        // If count is a null constant, the result should be all nulls.
+        context.moveOrCopyResult(
+            BaseVector::createNullConstant(outputType, numRows, pool),
+            rows,
+            result);
+        return;
+      }
+      count = args[1]->as<ConstantVector<int32_t>>()->valueAt(0);
+      checkCount(count);
+    } else {
+      exec::DecodedArgs decodedArgs(rows, args, context);
+      countDecoded = decodedArgs.at(1);
+    }
+
+    // Calculate the total number of the new elements.
+    auto totalCount = 0;
+    if (args[1]->isConstantEncoding()) {
+      totalCount = count * numRows;
+    } else {
+      rows.applyToSelected([&](vector_size_t row) {
+        auto count = countDecoded->isNullAt(row)
+            ? 0
+            : countDecoded->valueAt<int32_t>(row);
+        checkCount(count);
+        totalCount += count;
+      });
+    }
+
+    // Allocate new vectors for nulls, indices, lengths and offsets if
+    // necessary.
+    auto elementVector = args[0];
+    BufferPtr dictNulls;
```

Comment:
I don't think we need dictNulls. Do we?

Reply:
You are right - I removed the `dictNulls` and the code works just fine!!

I'm curious why we don't need to specify the nulls in the dict vector. Does the nulls have been referred from the BaseVector when it's being wrapped?

Reply:
We are just repeating values of the base vector. There is nothing special about repeating null values vs. non-null values.